Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/BoundsControlHandlesBase.cs

```diff
@@ -0,0 +1,258 @@
+                    configurationChanged.Invoke();
+                }
+            }
+        }
+
+        internal protected UnityEvent configurationChanged = new UnityEvent();
+        internal protected UnityEvent visibilityChanged = new UnityEvent();
+
+        internal void ResetHandleVisibility(bool isVisible)
+        {
+            if (handles != null)
+            {
+                for (int i = 0; i < handles.Count; ++i)
+                {
+                    handles[i].gameObject.SetActive(isVisible && IsVisible(handles[i]));
+                    BoundsControlVisualUtils.ApplyMaterialToAllRenderers(handles[i].gameObject, handleMaterial);
+                }
+            }
+        }
+
+        internal abstract bool IsVisible(Transform handle);
+        
+
+        internal protected List<Transform> handles = new List<Transform>();
+
+        public IReadOnlyList<Transform> Handles
+        {
+            get { return handles; }
+        }
+
+        internal void SetHighlighted(Transform handleToHighlight)
+        {
+            // turn off all handles that aren't the handle we want to highlight
+            if (handles != null)
+            {
+                for (int i = 0; i < handles.Count; ++i)
+                {
+                    if (handles[i] != handleToHighlight)
+                    {
+                        handles[i].gameObject.SetActive(false);
+                    }
+                    else
+                    {
+                        BoundsControlVisualUtils.ApplyMaterialToAllRenderers(handles[i].gameObject, handleGrabbedMaterial);
+                    }
+                }
+            }
+        }
+
+        internal void HandleIgnoreCollider(Collider handlesIgnoreCollider)
+        {
+            BoundsControlVisualUtils.HandleIgnoreCollider(handlesIgnoreCollider, handles);
+        }
+
+        internal void DestroyHandles()
+        {
+            if (handles != null)
+            {
+                foreach (Transform transform in handles)
+                {
+                    GameObject.Destroy(transform.gameObject);
+                }
+
+                handles.Clear();
+            }
+        }
+
+        internal bool IsHandleType(Transform handle)
+        {
+            for (int i = 0; i < handles.Count; ++i)
+            {
+                if (handle == handles[i])
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+
+        internal virtual HandleType GetHandleType()
+        {
+            return HandleType.None;
+        }
+
+        internal protected void SetMaterials()
+        {
+            if (handleMaterial == null)
+            {
+                float[] color = { 1.0f, 1.0f, 1.0f, 0.75f };
+
+                Shader shader = Shader.Find("Mixed Reality Toolkit/Standard");
+
+                handleMaterial = new Material(shader);
+                handleMaterial.EnableKeyword("_InnerGlow");
```

Comment:
This *might* not work on player builds due to the way shader permutations get built by Unity (unity won't include any shader permutations not referenced at build time). I would recommend we add a material reference as a serialized property somewhere and referencing it. Maybe in a MRTK profile?

Reply:
Yeah, it always surprises me when it works. I think since our profiles referenced enough materials we often get "lucky" with which shader variants get included.

Reply:
i created a resource folder for bounds control now that has a default material for handles - we fall back to that one if the user hasn't set up anything